compiler/internal/codegen/codegen_config_marshalers.go

```diff
@@ -254,11 +254,13 @@ func (cb *configUnmarshalersBuilder) readType(typ *schema.Type, pathElement Code
 	case *schema.Type_Config:
 		// The config type is the dynamic values which can be changed at runtime
 		// by unit tests
-		code, returnType := cb.readType(t.Config.Elem, pathElement)
 		if t.Config.IsValuesList {
-			return Qual("encore.dev/config", "CreateValueList").Call(code, Append(Id("path"), pathElement)), returnType
+			code, _ := cb.readType(t.Config.Elem, pathElement)
+			_, returnType := cb.readType(t.Config.Elem.GetList().Elem, pathElement)
```

Comment:
We need to seperate these, as we want the code of reading `[]T`, but we want the return type of `T` to put in the `config.Values[T]` type